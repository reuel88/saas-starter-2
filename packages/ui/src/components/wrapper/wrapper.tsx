import {
  ComponentPropsWithoutRef,
  ElementType,
  ForwardedRef,
  forwardRef,
} from "react";
import { cn } from "@repo/utilities";
import { BoxProps } from "../../types/box";

export const Wrapper = forwardRef(
  <T extends ElementType = "div">(
    {
      as,
      children,
      className = "",
      ...rest
    }: BoxProps<T> & Omit<ComponentPropsWithoutRef<T>, keyof BoxProps<T>>,
    ref: ForwardedRef<any>,
  ) => {
    const Component = as || "div";

    return (
      <Component
        ref={ref}
        className={cn("bg-background text-foreground", className)}
        {...rest}
      >
        {children}
      </Component>
    );
  },
);
