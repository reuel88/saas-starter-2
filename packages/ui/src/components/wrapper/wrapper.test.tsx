import { render, screen } from "@testing-library/react";
import { Wrapper } from "./wrapper";
import { expect } from "vitest";

describe("Wrapper", () => {
  it("should render a div element with children when no props are passed", () => {
    // Arrange

    // Act
    const { baseElement } = render(<Wrapper>Test</Wrapper>);

    // Assert
    expect(screen.getByText("Test")).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });
});
