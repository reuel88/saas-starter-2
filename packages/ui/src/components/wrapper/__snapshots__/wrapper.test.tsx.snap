// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Wrapper > should render a div element with children when no props are passed 1`] = `
<body>
  <div>
    <div
      class="bg-background text-foreground"
    >
      Test
    </div>
  </div>
</body>
`;
